syntax = "proto3";

package capsule;

import "capsule.proto";

message PairingRequest {

  repeated bytes sources = 1;
  repeated bytes sinks = 2;

  string replyAddr = 3;

  repeated bytes hashes = 4; // Used for base line anti-entropy
}

message PairingResponse {

  repeated CapsulePDU records = 1;

}

message PairingWrapperMsg {
  oneof msg_type {
      PairingRequest request = 1;
      PairingResponse response = 2;
  }   
}
